training/src/anemoi/training/commands/mlflow.py

```diff
@@ -131,5 +169,55 @@ def run(args: argparse.Namespace) -> None:
             ).sync()
             return
 
+        if args.subcommand == "prepare":
+            import mlflow
+            from hydra import compose
+            from hydra import initialize
+            from omegaconf import OmegaConf
+
+            from anemoi.training.diagnostics.mlflow.client import AnemoiMlflowClient
+
+            # Load configuration
+            with initialize(version_base=None, config_path="./"):
+                cfg = compose(config_name=args.config_name)
+
+            # Create MLflow client and get experiment
+            client = AnemoiMlflowClient(cfg.diagnostics.log.mlflow.tracking_uri, authentication=True)
+            experiment_id = client.get_experiment_by_name(cfg.diagnostics.log.mlflow.experiment_name).experiment_id
+
+            # Parse configuration
+            if cfg.training.run_id is not None:  # Existing run_id
+                LOGGER.info("Existing run_id: %s", cfg.training.run_id)
+                try:
+                    client.get_run(cfg.training.run_id)
+                except ValueError as e:
+                    msg = "Invalid run_id provided."
+                    raise ValueError(msg) from e
+                return
+
+            # Existing fork_id needs a new run_id attached to it
+            run = client.create_run(experiment_id, run_name=cfg.diagnostics.log.mlflow.run_name)
+            run_id = run.info.run_id
+            LOGGER.info("Creating new run_id: %s", run_id)
+
+            # Log the configuration file as an artifact
+            mlflow.set_tracking_uri(cfg.diagnostics.log.mlflow.tracking_uri)
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                fp = Path(tmp_dir, "config.json")
```

Comment:
some of this functionality is already written under the logger. py `log_hyperparams_as_artifact`.  I'd think we should think how to reuse it here (i.e logging the config as file and also we could log the config as hyperparameters)

Reply:
I added the function that logs the parameters in mlflow.utils.